bundles/org.openhab.transform.basicprofiles/src/test/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfileTest.java

```diff
@@ -731,6 +731,14 @@ public static Stream<Arguments> testFunctions() {
                 Arguments.of(decimalItem, "< 10%", decimals, DecimalType.valueOf("0.91"), true), //
                 Arguments.of(decimalItem, "< 10%", decimals, DecimalType.valueOf("0.89"), false), //
 
+                // Check against possible division-by-zero errors in $DELTA_PERCENT
+                Arguments.of(decimalItem, "> 10%", List.of(DecimalType.ZERO), DecimalType.valueOf("1"), true), //
+                Arguments.of(decimalItem, "< 10%", List.of(DecimalType.ZERO), DecimalType.valueOf("1"), true), //
+
+                // Check against possible division-by-zero errors in $DELTA_PERCENT
+                Arguments.of(decimalItem, "> 10%", List.of(DecimalType.ZERO), DecimalType.valueOf("1"), true), //
+                Arguments.of(decimalItem, "< 10%", List.of(DecimalType.ZERO), DecimalType.valueOf("1"), true), //
```

Comment:
I do not understand why the code is duplicated.

Reply:
.. and apropos my own comment .. perhaps you should test the cases UndefType.UNDEF resp. UndefType.NULL instead?

Reply:
> I do not understand why the code is duplicated.

@lolodomo it must've been copy paste error, or I was thinking of adding other tests and started with a copy to be modified but then I forgot to change it. Either way I don't remember which was the case, so for now I've removed the duplicates.




> .. and apropos my own comment .. perhaps you should test the cases UndefType.UNDEF resp. UndefType.NULL instead?

I've responded in your other comment